File: 백준/Silver/10828. 스택/스택.py
import sys
n = int(sys.stdin.readline())
stack = list()
for _ in range(n):
    a = sys.stdin.readline().split()
    if a[0] == 'push':
        b = a[1]
        stack.append(b)
    elif a[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif a[0] == 'size':
        print(len(stack))
    elif a[0] == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    else:
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[len(stack)-1])

File: 백준/Silver/10828. 스택/스택.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        ArrayDeque<Integer> stack = new ArrayDeque<>();
        for(int t = 0; t < T; t++) {
            String s = br.readLine();
            if (s.contains("push")) {
                String[] tmp = s.split(" ");
                stack.offer(Integer.parseInt(tmp[1]));
            }

            switch (s) {
                case "pop":
                    if (stack.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(stack.pollLast());
                    }
                    break;
                case "size":
                    System.out.println(stack.size());
                    break;
                case "empty":
                    if (stack.isEmpty()) {
                        System.out.println(1);
                    } else {
                        System.out.println(0);
                    }
                    break;
                case "top":
                    if (stack.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(stack.peekLast());
                    }
                    break;
            }
        }
    }
}